usr/lib/common/pkey_utils.c

```diff
@@ -546,14 +549,28 @@ CK_RV pkey_aes_ecb(STDLL_TokData_t *tokdata, SESSION *session,
 
     /* Call CPACF */
     memcpy(param.key, pkey_attr->pValue, pkey_attr->ulValueLen);
-    bytes_processed = s390_km(fc, &param, out_data, in_data, in_data_len);
-    if (bytes_processed <= 0) {
+    while (len > 0 && num_tries < PKEY_CONVERT_KEY_RETRIES) {
+        bytes_processed = s390_km(fc, &param, out_pos, in_pos, len);
+        if (bytes_processed < len) {
+            ret = convert_key(tokdata, session, key_obj, CK_FALSE,
```

Comment:
I just noticed that we do not see any trace messages when such a convert_key case happens. The convert_key routine does not itself issue any trace messages (unless an error occurs). It would be good to see this in the trace in case we ever need to debug an LGR case. So please add trace messages here before calling convert_key(), and at all other places where the convert function is called (also in the next 2 commits). You may include the bytes_processed and len variables into the trace message.

Reply:
ok, trace msgs are added.